app/Post_Types/Archive_Item/Rest.php

```diff
@@ -0,0 +1,343 @@
+<?php
+
+namespace Diviner\Post_Types\Archive_Item;
+
+use Carbon_Fields\Field;
+use Diviner\Post_Types\Diviner_Field\Diviner_Field;
+use Diviner\Post_Types\Diviner_Field\PostMeta as FieldPostMeta;
+use Diviner\Post_Types\Diviner_Field\Types\Text_Field;
+use Diviner\Post_Types\Diviner_Field\Types\Date_Field;
+use Diviner\Post_Types\Diviner_Field\Types\Taxonomy_Field;
+use Diviner\Post_Types\Diviner_Field\Types\CPT_Field;
+use Diviner\Post_Types\Diviner_Field\Types\Select_Field;
+use Diviner\Admin\Settings;
+use Diviner\CarbonFields\Helper;
+
+
+class Rest {
+
+	const FIELD_INDEX_ID = 'id';
+	const FIELD_INDEX_TITLE = 'title';
+	const FIELD_INDEX_TYPE = 'type';
+	const FIELD_INDEX_FIELD_ID = 'field_id';
+	const FIELD_INDEX_FIELD_POPUP = 'field_popup';
+
+	protected $container;
+	protected $fields;
+
+	public function init() {
+		$this->fields = $this->get_fields();
+	}
+
+	public function hooks() {
+		$action = 'after_setup_theme';
+		if ( DIVINER_IS_PLUGIN ) {
+			$action = 'plugins_loaded';
+		}
+		// add_filter( $action, [$this, 'init'], 10, 2 );
+		add_filter( 'rest_' . Archive_Item::NAME . '_query', [$this, 'rest_api_filter_add_vars'], 10, 2 );
```

Comment:
Here and elsewhere: smart IDEs like PHPStorm can look up dynamic filters like `"do_someting_with_{$target}"` but will not be able to do that with `'do_something_with_' . $target`.
Ideally modify the code to read like:
```php
class Foo{
	const BAR = 'bar';
}

$bar = Foo::BAR;
add_action("do_{$bar}_action", 'do_something');
```

This way IDEs, and users, will be able to search for the dynamic filter.

Reply:
Done